### ISSUE TITLE
Can not connect to node

### ISSUE THREAD

**Issue by Wangmmx** [docknetwork/sdk#223]
Is the node replaced or is there a problem with the node? Now the web socket nodes of the main network `wss://mainnet-node.dock.io` and testnet `wss://danforth-1.dock.io` cannot be connected

`Disconnected from wss://mainnet-node.dock.io: 1006:: connection failed`

**Comment by lovesh** [comment]
These endpoints should work. The explorer [here](https://fe.dock.io/#/explorer) uses these. However they do limit no of concurrent connections from an IP so see if you have some connections open already.

**Comment by Wangmmx** [comment]
Thanks for the reply, there is no connection anymore. I closed the connection and exited the pro. 

I used the local demo to test and reported an error. I attached my own code to another issue https://github.com/docknetwork/sdk/issues/222. 

The error in my pro is 
```
_Event {
  type: 'error',
  isTrusted: false,
  _yaeti: true,
  target:
   W3CWebSocket {
     _listeners: {},
     addEventListener: [Function: _addEventListener],
     removeEventListener: [Function: _removeEventListener],
     dispatchEvent: [Function: _dispatchEvent],
     _url: 'wss://mainnet-node.dock.io',
     _readyState: 3,
     _protocol: undefined,
     _extensions: '',
     _bufferedAmount: 0,
     _binaryType: 'arraybuffer',
     _connection: undefined,
     _client:
      WebSocketClient {
        _events: [Object: null prototype] {},
        _eventsCount: 0,
        _maxListeners: undefined,
        config: [Object],
        _req: null,
        protocols: [],
        origin: undefined,
        url: [Url],
        secure: true,
        base64nonce: 'fIwYDs7cIp8ia331eY53nQ==' },
     onclose: [Function: value],
     onerror: [Function: value],
     onmessage: [Function: value],
     onopen: [Function: value] },
  cancelable: true,
  stopImmediatePropagation: [Function] }
```
It seems to be a Polkadot related error. I have no problem running Polkadot tokens such as DOT and KSM. Only DCK will trigger this error. This code was successful the day before yesterday, I transferred DCK and the transfer was successfully and on chain. Have you modified anything related to the node? I didn't make any code changes. I just changed the transfer amount and suddenly the node reported an error, and I couldn't connect anymore.

**Comment by lovesh** [comment]
> The error in my pro is

This doesn't look like the complete error. 

> This code was successful the day before yesterday, I transferred DCK and the transfer was successfully and on chain. Have you modified anything related to the node?

Which commit were you using yesterday and which commit are you using now?

**Comment by Wangmmx** [comment]
I don't understand what you mean commit？

I used the same method today and the day before yesterday. I made an offline transaction the day before yesterday. The transaction hash is 0x03048c74e6994ef466eb46a09b444712d05bc714f81858c3ae1e534653bb564b. I connected the node to get some information such as nonce, balance, metadata, and now I just connected to the node and reported an error.

**Comment by lovesh** [comment]
By commit I mean the git commit hash of the SDK and node. 

Are you connecting to the mainnet or a testing with a local node?

We pushed a small runtime upgrade to the node yesterday which caused the change in metadata. You can see both the node and sdk releases here https://github.com/docknetwork/dock-substrate/releases and here https://github.com/docknetwork/sdk/releases

**Comment by Wangmmx** [comment]
I didn't use the SDK, because I did polkadot's offline signature before, so I used the compatible logic directly.

I am connected to an external node `wss://mainnet-node.dock.io`
I will take a look at the releases first. These two issues (222 and 223)actually mean the same thing. Because the connected websocket reported an error, I wrote a few lines of demo to test, but I can’t connect. Thank you for replying to me. I see that the browser is normal.But I don’t know why I just keep reporting error, it bothers me a lot

**Comment by lovesh** [comment]
I cant tell you the exact reason but maybe close the browser while running the script as the total number of connections from your IP might be exceeding.

**Comment by Wangmmx** [comment]
I switched the IP and it didn’t work, it’s so weird, thank you for helping me, let me see it again

**Comment by Wangmmx** [comment]
I tried to change the connection method, but the latest SDK connection was unsuccessful.
![image](https://user-images.githubusercontent.com/22836587/103983392-20281300-51c0-11eb-8d74-c2198781a866.png)

Could you please tell me that my ip is limited by the number of connections. How can I solve this?

**Comment by Wangmmx** [comment]
Sorry to bother you again, I still can't connect to the node, I am doing DCK exchange support here, because I can't find the reason, I am a little more anxious

I have seen the releases you mentioned, and they don’t seem to have an impact on me

The test code has been shown to you. I can’t connect here. It’s okay on your side. You also told me that the node is okay. What do you mean when the number of ip connections is restricted? Can i solve it?

**Comment by lovesh** [comment]
> What do you mean when the number of ip connections is restricted?

wss://mainnet-node.dock.io doesn't allow more than 5 connections from a single IP.

**Comment by SamHellawell** [comment]
Hey @Wangmmx  - this error seems pretty odd. So, with PolkadotJS and our SDK I can connect to the mainnet node. can you try visiting http://mainnet-node.dock.io/ directly and let me know the resuit. You should see something like "WebSocket Protocol Error: Unable to parse WebSocket key."

If not, its a deeper issue with your network.

**Comment by Wangmmx** [comment]
> > What do you mean when the number of ip connections is restricted?
> 
> wss://mainnet-node.dock.io doesn't allow more than 5 connections from a single IP.

I tried to change the ip but couldn’t connect, thanks for the reply, I couldn’t reply you in time due to the time difference. Sorry

**Comment by Wangmmx** [comment]
> Hey @Wangmmx - this error seems pretty odd. So, with PolkadotJS and our SDK I can connect to the mainnet node. can you try visiting http://mainnet-node.dock.io/ directly and let me know the resuit. You should see something like "WebSocket Protocol Error: Unable to parse WebSocket key."
> 
> If not, its a deeper issue with your network.

Thank you very much for your reply. I have used polkadot's SDK and npm libraries before. Similar tokens such as DOT, KSM, and PCX can be connected normally (it is still connected and continue to send transactions), only the DCK connection is interrupted and not connected successfully anymore.


 Is it a direct jump? I get this page
![image](https://user-images.githubusercontent.com/22836587/104080367-6d5cc100-5262-11eb-90e4-b35cb0677c8c.png)

**Comment by SamHellawell** [comment]
That's an interesting error, not sure why nginx would 503 for you but works just fine for me. I wonder, could you try incognito mode just to see if any extensions are interfering potentially? Do you have any specific network configuration?

**Comment by Wangmmx** [comment]
After I opened the network proxy, I can get the error you said, "WebSocket Protocol Error: Unable to parse WebSocket key.",but the connection nodes still report errors. Neither my colleague nor I can connect successfully. Finally, we put the project online, the connection was success and the transaction was able to be sent. 

I don’t know the specific reason yet, but I closed the issue first, and I will continue to follow up.
